tests/NLog.UnitTests/Targets/ColoredConsoleTargetTests.cs

```diff
@@ -119,6 +119,80 @@ public void WordHighlightingRegex(bool compileRegex)
                 new string[] { "The ", "cat", " ", "sat", " at the bar." });
         }
 
+        [Fact]
+        public void ColoredConsoleAnsi_OverlappingWordHighlight_VerificationTest()
+        {
+            var target = new ColoredConsoleTarget { Layout = "${logger} ${message}", EnableAnsiOutput = true };
+            target.UseDefaultRowHighlightingRules = false;
+            target.WordHighlightingRules.Add(new ConsoleWordHighlightingRule
+            {
+                Text = "big warning",
+                ForegroundColor = ConsoleOutputColor.DarkRed,
+                BackgroundColor = ConsoleOutputColor.NoChange
+            });
+            target.WordHighlightingRules.Add(new ConsoleWordHighlightingRule
+            {
+                Text = "warn",
+                ForegroundColor = ConsoleOutputColor.DarkMagenta,
+                BackgroundColor = ConsoleOutputColor.NoChange
+            });
+            target.WordHighlightingRules.Add(new ConsoleWordHighlightingRule
+            {
+                Text = "a",
+                ForegroundColor = ConsoleOutputColor.DarkGreen,
+                BackgroundColor = ConsoleOutputColor.NoChange
+            });
+
+            AssertOutput(target, "The big warning message",
+                    new string[] { "The \x1B[31mbig \x1B[35mw\x1B[32ma\x1B[35mrn\x1B[31ming\x1B[0m mess\x1B[32ma\x1B[0mge\x1B[0m" });
+        }
+
+        [Fact]
+        public void ColoredConsoleAnsi_RepeatedWordHighlight_VerificationTest()
+        {
+            var target = new ColoredConsoleTarget { Layout = "${logger} ${message}", EnableAnsiOutput = true };
+            target.UseDefaultRowHighlightingRules = false;
+            target.WordHighlightingRules.Add(new ConsoleWordHighlightingRule
+            {
+                Text = "big big",
+                ForegroundColor = ConsoleOutputColor.DarkRed,
+                BackgroundColor = ConsoleOutputColor.NoChange
+            });
+
+            AssertOutput(target, "The big big big big warning message",
+                    new string[] { "The \x1B[31mbig big\x1B[0m \x1B[31mbig big\x1B[0m warning message\x1B[0m" });
+        }
+
+        [Fact]
+        public void ColoredConsoleAnsi_RowColor_VerificationTest()
+        {
+            var target = new ColoredConsoleTarget { Layout = "${message}", EnableAnsiOutput = true };
+            target.UseDefaultRowHighlightingRules = false;
+            target.RowHighlightingRules.Add(new ConsoleRowHighlightingRule() { BackgroundColor = ConsoleOutputColor.DarkGreen });
+
+            AssertOutput(target, "The big warning message",
+                    new string[] { "\x1B[42mThe big warning message\x1B[0m" },
+                    string.Empty);
+        }
+
+        [Fact]
+        public void ColoredConsoleAnsi_RowColorWithWordHighlight_VerificationTest()
+        {
+            var target = new ColoredConsoleTarget { Layout = "${message}", EnableAnsiOutput = true };
```

Comment:
tests looks good. 

You please create a method to deduplicate the tests a bit? Then I will merge it. Thanks!

Reply:
What tests do you find problematic? I find the tests very simple

Reply:
maybe create a method for 

```
            var target = new ColoredConsoleTarget { Layout = "${logger} ${message}", EnableAnsiOutput = true };
            target.UseDefaultRowHighlightingRules = false;
            target.WordHighlightingRules.Add(new ConsoleWordHighlightingRule
            {
                Text = "big big",
                ForegroundColor = ConsoleOutputColor.DarkRed,
                BackgroundColor = ConsoleOutputColor.NoChange
            });

```

etc? It's duplicated now

Reply:
I find it obfuscating as one includes Row-Highlight, the other does not. Two different tests.